packages/shared/src/utils/files/file-uploader-utils.ts

```diff
@@ -65,22 +42,31 @@ export const compressImageFiles = (files: TFile[]) => {
     return Promise.all(promises);
 };
 
-export const readFiles = (files: TFile[], getFileReadErrorMessage: (t: string) => string, settings: TSettings) => {
-    const promises: Promise<TFileObject | { message: string }>[] = [];
+export const readFiles = (
+    files: Blob[],
+    getFileReadErrorMessage: (t: string) => string,
+    settings?: Partial<TSettings>
+) => {
+    const promises: Array<Promise<Partial<TFileObject> | { message: string }>> = [];
 
     files.forEach(f => {
         const fr = new FileReader();
-        const promise = new Promise<TFileObject | { message: string }>(resolve => {
+        const promise = new Promise<Partial<TFileObject> | { message: string }>(resolve => {
             fr.onload = () => {
-                const file_obj = {
+                const file_metadata = {
                     filename: f.name,
                     buffer: fr.result,
                     documentFormat: getFormatFromMIME(f),
                     file_size: f.size,
-                    ...settings,
-                    documentType: settings?.documentType ?? 'utility_bill',
+                    documentType: settings?.document_type ?? DOCUMENT_TYPES.utility_bill,
+                    documentId: settings?.document_id,
+                    expirationDate: settings?.expiration_date,
+                    lifetimeValid: settings?.lifetime_valid,
+                    pageType: settings?.page_type,
+                    proof_of_ownership: settings?.proof_of_ownership,
+                    document_issuing_country: settings?.document_issuing_country,
                 };
-                resolve(file_obj);
+                resolve(file_metadata);
             };
 
             fr.onerror = () => {
```

Comment:
should we resolve the error or reject?

Reply:
This is expected to resolve as the error is being handled with the key `message`